adapters/socketio_win32.c

```diff
@@ -324,35 +364,37 @@ CONCRETE_IO_HANDLE socketio_create(void* io_create_parameters)
                 if ((result->hostname == NULL) && (result->socket == INVALID_SOCKET))
                 {
                     LogError("Failure: hostname == NULL and socket is invalid.");
-                    singlylinkedlist_destroy(result->pending_io_list);
-                    free(result);
+                    destroy_socket_io_instance(result);
+                    result = NULL;
+                }
+                else if ((result->addrInfo = calloc(1, sizeof(struct addrinfo))) == NULL)
+                {
+                    LogError("Failure: addrInfo == NULL.");
+                    destroy_socket_io_instance(result);
+                    result = NULL;
+                }
+                else if ((result->addrInfo->ai_addr = calloc(1, sizeof(struct sockaddr_in))) == NULL)
+                {
+                    LogError("Failure allocating ai_addr");
+                    destroy_socket_io_instance(result);
+                    result = NULL;
+                }
+                else if ((result->dns_resolver = dns_resolver_create(result->hostname, result->port, NULL)) == NULL)
```

Comment:
>port [](start = 95, length = 4)

port =0? it should be config->port right?

Reply:
You are absolutely right. Fixed.
Thanks, you saved me here.